DigitalLearningSolutions.Data/DataServices/SupervisorService.cs

```diff
@@ -699,10 +699,29 @@ FROM    CandidateAssessments
                new { selfAssessmentId, delegateUserId });
             if (candidateAssessmentId > 0)
             {
-                int numberOfAffectedRows = connection.Execute(
-                    @"INSERT INTO CandidateAssessmentSupervisors (CandidateAssessmentID, SupervisorDelegateId, SelfAssessmentSupervisorRoleID)
-                            VALUES (@candidateAssessmentId, @supervisorDelegateId, @selfAssessmentSupervisorRoleId)", new { candidateAssessmentId, supervisorDelegateId, selfAssessmentSupervisorRoleId }
-                    );

```

Comment:
@Swapnamol You should be fine to just remove your sql insert here (ie no code changes to this file).
The changes I made in TD-1079 now first check for an existing record then perform an Update, otherwise it does an Insert as per the original code (ie your lines 702 to 705 have just moved down to lines 720 to 724 in the uar-test branch).